### ISSUE TITLE
Add the PasswordEncoder process

### ISSUE THREAD

**Issue by NABEEL-AHMED-JAMIL** [bfwg/angular-spring-starter#15]
Grate Work..... kindly add the PasswordEncoder I try to added but there are some issue/error show. may be due to the your default user name and passward store in the database.
I also try to PasswordEncoder in side the your test service from where you put the data to the database
but it still show the same error (PasswordEncoder Bean) and (Authentication size 0)

**Comment by NABEEL-AHMED-JAMIL** [comment]
And one think thank again for this post

**Comment by bfwg** [comment]
@NABEEL-AHMED-JAMIL Thanks, I'll try to add it sometime soon.

**Comment by bfwg** [comment]
@NABEEL-AHMED-JAMIL The password encoder has been added. 🎉

**Comment by NABEEL-AHMED-JAMIL** [comment]
Grate Work Mr.Fan Jin. I also think this same way to resolve this problem
of password encoder. one think in this post basically user send the user
name and password
to get the auth tokent form the user. and the token send to get the user
info. is their any way we get the user info with the token like the json
below i show you here.
{
    token: {lkdfjlsja24ksjfo2342232r4}
    account: {
                      firstname: xyz
                      lastname: xyz
                      username: username
                      email: xyx@gmail.com
                      active: 0/1
                      // password not send to the front-end side i now the
way the password not send either by adding the null value in the user
detail service or either in this you example you used the jsonIgnore
                   }
}
so basically thanks for this post and one think is this any way to do this
upper josn form. I'm not asking you to give me the solve problem. I'm just
Questioning you. Hope not mind and this one for you👍
message from :- Nabeel Ahmed

**Comment by bfwg** [comment]
Hi @NABEEL-AHMED-JAMIL , for a custom token response, you need to define those variables in `UserTokenState ` class, since we are using `UserTokenState` as the login success response object. 
https://github.com/bfwg/angular-spring-starter/blob/master/server/src/main/java/com/bfwg/security/auth/AuthenticationSuccessHandler.java#L54

**Comment by NABEEL-AHMED-JAMIL** [comment]
Hello Fan Jin Hope you remember me. I'm follow your spring and angular 2
project when i send the request from my code it show this error

zone.js:2263 OPTIONS http://localhost:8080/auth/login 403 ()
XMLHttpRequest cannot load http://localhost:8080/auth/login. Response to
preflight request doesn't pass access control check: No
'Access-Control-Allow-Origin' header is present on the requested resource.
Origin 'http://localhost:4200' is therefore not allowed access. The
response had HTTP status code 403.

I apply the Cross Filter both side. but same response get any idea how to
solve this problem
one think when i send the request from the your project it work find i was
try many request of my project mapping from the user project they work find
when i send from the my project it give the above error

**Comment by bfwg** [comment]
Hey @NABEEL-AHMED-JAMIL , have you configured `proxy.conf` in your frontend project?
https://github.com/bfwg/angular-spring-starter/blob/master/frontend/proxy.conf.json#L3

**Comment by NABEEL-AHMED-JAMIL** [comment]
yes But same result i found this is my github where i store my project with
the name of Hospital Front and Backend.
The proxy linke is
https://github.com/NABEEL-AHMED-JAMIL/HospitalFrontEnd/blob/master/proxy.conf.json
and the project link is https://github.com/NABEEL-AHMED-JAMIL

**Comment by bfwg** [comment]
Hi @NABEEL-AHMED-JAMIL , I think something is going wrong in you https://github.com/NABEEL-AHMED-JAMIL/HospitalBackEnd/blob/master/src/main/java/com/ballistic/hospital/config/WebSecurityConfig.java#L74 , but I can't tell with a quick glance. More investigation is needed.

**Comment by NABEEL-AHMED-JAMIL** [comment]
well tank's for the reply and investigate my project. when i send the
request from you login form it give me the token and my info object. but in
my form when i send the post/request method for calling the end-point it
show the error

**Comment by NABEEL-AHMED-JAMIL** [comment]
well I'm trying to handle this problem. if found I will share you the
solution. Thanks for the such a good replay's of all message By.
👍

**Comment by bfwg** [comment]
No problem, keep me in the loop.

**Comment by NABEEL-AHMED-JAMIL** [comment]
ok

**Comment by NABEEL-AHMED-JAMIL** [comment]
are you work on freelancer

**Comment by bfwg** [comment]
H, @NABEEL-AHMED-JAMIL , no, I work on a full-time job.

**Comment by NABEEL-AHMED-JAMIL** [comment]
good i think found the problem check this file
https://github.com/NABEEL-AHMED-JAMIL/HospitalFrontEnd/blob/master/src/app/_services/config.service.ts#L8

**Comment by NABEEL-AHMED-JAMIL** [comment]
Well i resolve my issue by using this code
private serializeObj(obj) {
var result = [];
for (var property in obj)
result.push(encodeURIComponent(property) + "=" + encodeURIComponent(obj[
property]));
return result.join("&");
}

// login service
public login(loginDto: any): Observable<any> {
let headers = new Headers({ 'Content-Type': 'application/x-www-form-urlencoded;
charset=UTF-8' });
let options = new RequestOptions( {method: RequestMethod.Post, headers:
headers });
let body = this.serializeObj(loginDto);
//debugger;
return this.http.post(this.configService.getlogin_url, body, options )
.map((response: Response) => {
console.log(response);
const Uresponse = response.json();
if (Uresponse.access_token !== null) {
localStorage.setItem('currentUser', JSON.stringify(Uresponse.doctor));
localStorage.setItem('access_token' , JSON.stringify(response['access_token'
]));
}
return response.json();
});
}

**Comment by NABEEL-AHMED-JAMIL** [comment]
hope not mind i send this message as you asked me keep me in the loop that
why i send this message to you and one think i like your work that you post
on the git.
and thanks for the sharing the spring-boot-auth.
By
😀

**Comment by bfwg** [comment]
No problem dude, I'm glad it worked out for you.

**Comment by NABEEL-AHMED-JAMIL** [comment]
hello Fan hope you remember me. i see your last update password rest and
other new changes good work. can you tell me how can i upload my
application like your app demo
